tensorflow/python/ops/array_ops.py

```diff
@@ -1132,16 +1163,32 @@ def boolean_mask(tensor, mask, name="boolean_mask", axis=None):
   # 1-D example
   tensor = [0, 1, 2, 3]
   mask = np.array([True, False, True, False])
-  boolean_mask(tensor, mask)  # [0, 2]
+  tf.boolean_mask(tensor, mask).numpy()  
+  # ==> array([0, 2], dtype=int32)
   ```
 
-  In general, `0 < dim(mask) = K <= dim(tensor)`, and `mask`'s shape must match
-  the first K dimensions of `tensor`'s shape.  We then have:
+  In general, `0 < tf.rank(mask) = K <= tf.rank(tensor)`, and `mask`'s shape 
+  must match the first K dimensions of `tensor`'s shape.  We then have:
     `boolean_mask(tensor, mask)[i, j1,...,jd] = tensor[i1,...,iK,j1,...,jd]`
   where `(i1,...,iK)` is the ith `True` entry of `mask` (row-major order).
+  For example:
+
+  ```python
+  # 3-D tensor and 2-D mask
+  tensor = [[[1, 2],
+             [3, 4]],
+            [[5, 6],
+             [7, 8]]]
+  mask = [[True, False],
+          [False, True]]
+  tf.boolean_mask(tensor, mask).numpy()
```

Comment:
Similarly, will `numpy()` work?

Reply:
Yes, this code snippet is also intended for eager mode. Added a note to the comment at the top of the snippet.